### PR TITLE
Fix remaining `vue-compat` warnings

### LINKED ISSUE #4907: [Linting] Investigate option chaining linting rule generated by eslint
#### Summary
There is a rule being thrown about option chaining from eslint when running the following command:
```
npm run lint
```
We should investigate and remove/update/suppress the rule.

```
> openmct@2.0.1-SNAPSHOT lint
> eslint example src --ext .js,.vue openmct.js


/home/runner/work/openmct/openmct/src/api/objects/MutableDomainObject.js
Warning:   9[6](https://github.com/nasa/openmct/runs/5397657377?check_suite_focus=true#step:6:6):89  warning  Consider using the native optional chaining to get nested values and nullish coalescing operator for fallback values  you-dont-need-lodash-underscore/get

/home/runner/work/openmct/openmct/src/plugins/plot/configuration/PlotConfigurationModel.js
Warning:   133:32  warning  Consider using the native optional chaining to get nested values and nullish coalescing operator for fallback values  you-dont-need-lodash-underscore/get
Warning:   134:33  warning  Consider using the native optional chaining to get nested values and nullish coalescing operator for fallback values  you-dont-need-lodash-underscore/get

/home/runner/work/openmct/openmct/src/plugins/plot/configuration/YAxisModel.js
Warning:   1[7](https://github.com/nasa/openmct/runs/5397657377?check_suite_focus=true#step:6:7)0:23  warning  Consider using the native optional chaining to get nested values and nullish coalescing operator for fallback values  you-dont-need-lodash-underscore/get

/home/runner/work/openmct/openmct/src/ui/toolbar/Toolbar.vue
Warning:   1[8](https://github.com/nasa/openmct/runs/5397657377?check_suite_focus=true#step:6:8)4:[20](https://github.com/nasa/openmct/runs/5397657377?check_suite_focus=true#step:6:20)  warning  Consider using the native optional chaining to get nested values and nullish coalescing operator for fallback values  you-dont-need-lodash-underscore/get
```

### LINKED ISSUE #7124: Fix remaining `vue-compat` warnings
#### Summary

In order to finish migrating to Vue 3 proper, we will need to fix all of the remaining warnings generated from `vue-compat`.

### PR DESCRIPTION
**Pull request by depperm**
Closes #7124 

### Describe your changes:

- Fix remaining `vue-compat` warnings by:
  - add emits in declaration
  - file components are PascalCase and MultiWord
  - added eventbus for two places use `$once` and `$on`, (might be a better way to handle this)
  - remove use of `$listeners` as it is deprecated
  - remove deprecated `Vue` import in favor of importing individual methods

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/api/forms/components/controls/CompositeContainer.vue
Comment (depperm) [inline src/api/forms/components/controls/CompositeContainer.vue:1]: I don't think this component is used....which means CompositeItem isn't either.

**Comment by ozyx** [comment]
@depperm This is awesome! This will help in our migration to Vue 3 proper. Before we can merge, we will need the tests to be passing. It looks like the e2e tests are failing due to file path changes or something similar. Could you please look into and fix the failing tests on #6955 ? Please let us know if you have any questions about it, we are happy to help!

**Comment by depperm** [comment]
@ozyx Looking at the tests that are failing: 
```
...
ERROR in ./src/ui/layout/search/GrandSearch.vue?vue&type=script&lang=js (./node_modules/vue-loader/dist/index.js??ruleSet[0]!./src/ui/layout/search/GrandSearch.vue?vue&type=script&lang=js) 2:0-49
Module not found: Error: Can't resolve '../../components/Search.vue' in '/root/project/src/ui/layout/search'

ERROR in ./src/plugins/LADTable/LADTableConfigurationViewProvider.js 23:0-84
Module not found: Error: Can't resolve './components/LadTableConfiguration.vue' in '/root/project/src/plugins/LADTable'

ERROR in ./src/plugins/LADTable/LADTableView.js 23:0-49
Module not found: Error: Can't resolve './components/LadTable.vue' in '/root/project/src/plugins/LADTable'

ERROR in ./src/plugins/clearData/plugin.js 23:0-73
Module not found: Error: Can't resolve './components/GlobalClearIndicator.vue' in '/root/project/src/plugins/clearData'

ERROR in ./src/plugins/flexibleLayout/flexibleLayoutViewProvider.js 24:0-70
Module not found: Error: Can't resolve './components/FlexibleLayout.vue' in '/root/project/src/plugins/flexibleLayout'

ERROR in ./src/plugins/tabs/tabs.js 24:0-50
Module not found: Error: Can't resolve './components/Tabs.vue' in '/root/project/src/plugins/tabs'

ERROR in ./src/plugins/telemetryTable/TelemetryTableView.js 1:0-52
Module not found: Error: Can't resolve './components/Table.vue' in '/root/project/src/plugins/telemetryTable'
```

it doesn't look like the file names are updating. I'm not sure how to fix this. Looking at the last example it says it can't resolve Table.vue but I've updated the code to TableComponent.vue:
file: https://github.com/nasa/openmct/pull/6966/files#diff-09db24f148ad1c2ff6dc49bf89aa41d0cb53de9d8062217a99ac75dd77d35440

I can't find any reference to Table.vue

**Comment by depperm** [comment]
#6988 might resolve

**Comment by ozyx** [comment]
@depperm could you fix the conflicts here please? Thanks!

**Comment by ozyx** [comment]
@depperm Thanks for doing the heavy lifting on this one! I'm going to have another member of the team give this a re-review before we merge because it's so large, but this PR gets us to the point where we will be able to upgrade to Vue 3 proper. That's huge! Well done.

**Review by scottbell — APPROVED** [review]
Did some cursory testing of Open MCT, particularly with plots and configurations, and it all looked good! On inspection, code looks straightforwardly changed too.

**Pull request merged**